test/lib/rules/yield-effects-spec.js

```diff
@@ -25,9 +79,73 @@ ruleTester.run("yield-effects", rule, {
   ],
   invalid: [
     {
+      code: "import { take } from 'redux-saga';"
+          + "export default function* saga() {"
+          + "  const sagaTakeEffect = take();"
+          + "  yield sagaTakeEffect;"
+          + "}",
+      errors: [ERRORS.noyield]
+    },
+    {
+      code: "import { take as sagaTake } from 'redux-saga';"
+          + "export default function* saga() {"
+          + "  const sagaTakeEffect = sagaTake();"
+          + "  yield sagaTakeEffect;"
```

Comment:
Another plugin that would deal with generators should treat this code as valid, because it is. You can yield anything.

But, when using `redux-saga`, you should only yield the return values of the effect creators (or arrays of such).

So, it's this `eslint-plugin-redux-saga`'s responsibility to limit the usage of generators to only yield `redux-saga` effects when a saga file is detected (by assuming files which import `redux-saga` as files that declare sagas, and generators inside them are sagas).